t/sequences.t

```diff
@@ -266,7 +266,112 @@ FASTA
+CHSPYTSPAAHWSSPGASPYLKQPPALTPSSNPAASAGLHSSMSSYSLEQSYLHQNARED
+LSVGLPRYQHHSTPVCDRKDFVLNFNGISSFHPSASGSYYHHHHQSVCQDIKPCVM
+FASTA
+  is($fasta, $expected, 'Getting protein sub-sequence from transcript without end parameter');
+}
+{
+  my $id = 'ENSP00000259806';
+  my $url = "/sequence/id/$id?type=protein&start=10&end=30";
+  my $fasta = fasta_GET($url, 'Getting protein sub-sequence from transcript without end parameter');
+  my $expected = <<'FASTA';
+>ENSP00000259806
+APLRRACSPVPGALQAALMSP
+FASTA
+  is($fasta, $expected, 'Getting protein sub-sequence from transcript without end parameter');
+}
+{
+  my $id = 'ENST00000400701';
+  my $url = "/sequence/id/$id?type=protein&end=200";
+  my $fasta = fasta_GET($url, 'Getting protein sub-sequence from transcript without start parameter');
+  my $expected = <<'FASTA';
+>ENSP00000383537
+XSNLKRDVAHLYRGVGSRYIMGSGESFMQLQQRLLREKEAKIRKALDRLRKKRHLLRRQR
+TRREFPVISVVGYTNCGKTTLIKALTGDAAIQPRDQLFATLDVTAHAGTLPSRMTVLYVD
+TIGFLSQLPHGLIESFSATLEDVAHSDLILHVRDVSHPEAELQKCSVLSTLRGLQLPAPL
+LDSMVEVHNKVDLVPGYSPTEPNVVPVSALRGHGLQELKAELDAAVLKATGRQILTLRVR
+LAGAQLS
+FASTA
+  is($fasta, $expected, 'Getting protein sub-sequence from transcript without start parameter');
+}
+{
+  my $id = 'ENSG00000112699';
+  my $url = "/sequence/id/$id?type=protein&multiple_sequences=1&start=300000";
+  my $fasta = fasta_GET($url, 'Getting protein sub-sequence from gene, using multiple');
+  my $expected = <<'FASTA';
+>ENSP00000436726
+AMWLMLQNDEPEDFVIATGEVHSVREFVEKSFLHIGKTIVWEGKNENEVGRCKETGKVHV
+TVDLKYYRPTEVDFLQGDCTKAKQKLNWKPRVAFDELVREMVHADVELMRTNPNA
+>ENSP00000370194
+GANFVTRKISRSVAKIYLGQLECFSLGNLDAKRDWGHAKDYVEAMWLMLQNDEPEDFVIA
+TGEVHSVREFVEKSFLHIGKTIVWEGKNENEVGRCKETGKVHVTVDLKYYRPTEV
+FASTA
+  is($fasta, $expected, 'Getting protein sub-sequence from gene, using multiple');
+}
+{
+  my $id = 'ENST00000259806';
+  my $url = "/sequence/id/$id?type=protein&end=100";
+  action_check_code($url, 400, 'Return code for out of boundaries parameters should be 400');
+  action_raw_bad_regex($url, qr/not within the sequence/, 'There\'s no protein sequence contained in the genomic boundaries given');
+}
+{
+  my $id = 'ENSG00000243439';
+  my $url = "/sequence/id/$id?start=1000&content-type=application/json";
+  action_check_code($url, 400, 'Return code for out of boundaries parameters should be 400');
+  action_raw_bad_regex($url, qr/not within the sequence/, 'Start of sub-sequence range can not be beyond the sequence length');
+}
+{
+  my $url = "/sequence/id/?start=120&end=150";
+  my $body = q/{ "ids" : [ "ENSP00000370194", "ENSG00000243439" ]}/;
+  my $seq_a = q/LAEYTADVDGVGTLRLLDAVKTCGLINSVKF/;
+  my $seq_b = q/TGCACTAAGTTCAGCATGAAGAGCAGCGGGC/;
+  my $response = [{"desc" => undef,"id" => "ENSP00000370194","seq" => $seq_a,"molecule" => "protein"},{"desc" => "chromosome:GRCh37:6:1507676:1507706:1","id" => "ENSG00000243439","seq" => $seq_b,"molecule" => "dna"}];
+  is_json_POST($url,$body,$response,'POST ID sequence fetch with sequence trimming');
 
```

Comment:
I'd like to see a test that probes the behaviour of a start = end subseq, for a single base.